lib/MySQL_Monitor.cpp

```diff
@@ -872,10 +874,12 @@ void * monitor_connect_thread(void *arg) {
 			(strncmp(mmsd->mysql_error_msg,"ProxySQL Error: Access denied for user",strlen("ProxySQL Error: Access denied for user"))==0)
 		) {
 			proxy_error("Server %s:%d is returning \"Access denied\" for monitoring user\n", mmsd->hostname, mmsd->port);
+			MyHGM->p_update_mysql_error_counter(p_mysql_error_type::proxysql, mmsd->hostgroup_id, mmsd->hostname, mmsd->port, ER_ACCESS_DENIED_ERROR);
```

Comment:
Check for logging error is being performed on `mmsd->mysql_error_msg`, I assumed that if `mysql_errno(mmsd->mysql)` was guaranteed to be ER_ACCESS_DENIED_ERROR this check would much more simpler.

Reply:
Because those are the only messages we want to write in the error log.
I don't see a valid reason to hardcode an error code, even if we know already what is.